### ISSUE TITLE
RangeError when saving a buffer

### ISSUE THREAD

**Issue by kay-is** [neumino/thinky#416]
I get this error:

    [RangeError: out of range index]

I have this model:

    const MyModel = thinky.createModel('myModel', {
      id: type.string(),
      label: type.string(),
      image: type.buffer().optional(),
    })

I get this error on a `save()` call after I used `merge()`

    const image = Buffer.concat(fileUploadParts)
    myDocument.merge({image}).save().then(...).catch(e => console.log(e))

This error seem to relate to a wrong use length somewhere.

When I use `image.toString('base64')` it produces a valid base64 string that can be converted to a png without errors, so the buffer seems to be okay.

Also, this works:

    const carImage = new Buffer('iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEwAACxMBAJqcGAAAAM5JREFUWIXtk0EKwjAQRV9FUCjoVryGHiAeQCl4oXonDxAX7vQconbbdV1khGJLSdpAQeZBoAyTP48wBUVRFEUJwwBV4MljCtgeAtEkTM/h0STsQIEgiTVwAE7AGXhEGP49FveaACQybAtsamflazqAC5AnYjUakzGHdwlcgR2QAkvcPtwDs70zfpfkBsxa+uYS6rNoIRmNy3tpPgIF8AYyqRlPgZCMxuWFNBa12ktqqaeAd8YU9yu2UbV8lx390TMynPUTt0B9iJGhKH/OB7KGpL3XEmenAAAAAElFTkSuQmCC', 'base64')

    const document = new MyModel({
      customerId: customer.id,
      label: 'Vehicle',
      image: carImage,
    })
    document.save()

So saving a buffer to RethinkDB doesn't seem to be an issue in general.

**Comment by neumino** [comment]
What do you pass to merge? How does `{image}` work?

**Comment by kay-is** [comment]
I pass  `{image: image}` to merge and image is a buffer.

I found that the error is happening here: 

https://github.com/neumino/rethinkdbdash/blob/master/lib/term.js#L2382

Seems like the `.toString('base64')` call fail there. But when I call this manually on my buffer, before passing it into `.merge()` I don't get the error.

**Comment by neumino** [comment]
Can you provide a script that trigger this error? This works fine for me:
```
const carImage = new Buffer('iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEwAACxMBAJqcGAAAAM5JREFUWIXtk0EKwjAQRV9FUCjoVryGHiAeQCl4oXonDxAX7vQconbbdV1khGJLSdpAQeZBoAyTP48wBUVRFEUJwwBV4MljCtgeAtEkTM/h0STsQIEgiTVwAE7AGXhEGP49FveaACQybAtsamflazqAC5AnYjUakzGHdwlcgR2QAkvcPtwDs70zfpfkBsxa+uYS6rNoIRmNy3tpPgIF8AYyqRlPgZCMxuWFNBa12ktqqaeAd8YU9yu2UbV8lx390TMynPUTt0B9iJGhKH/OB7KGpL3XEmenAAAAAElFTkSuQmCC', 'base64')
```

**Comment by kay-is** [comment]
Funny thing is, what you posted worked for me too (I already had this in my code to seed the DB)

What didn't work, was:

1. Uploading the image with `multipart/form-data` from a HTTP client to the server. (works)
2. Getting the file stream with busboy (works)
3. Reading the stream and concatenating it's parts with `Buffer` (works)
4. Calling `toString('base64')` on the buffer to test that I got the right data (works)
5. Merging the buffer to the document I got from the thinky model (works)
6. Saving the document (error on `toString('base64')` in rethinkdbdash)
